components/feature/session/src/main/java/mozilla/components/feature/session/prompt/PromptFeature.kt

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session.prompt
+
+import android.support.annotation.VisibleForTesting
+import android.support.v4.app.FragmentManager
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.prompt.Choice
+import mozilla.components.concept.engine.prompt.PromptRequest
+import mozilla.components.concept.engine.prompt.PromptRequest.MultipleChoice
+import mozilla.components.concept.engine.prompt.PromptRequest.SingleChoice
+import mozilla.components.concept.engine.prompt.PromptRequest.MenuChoice
+import mozilla.components.feature.session.prompt.ChoiceDialogFragment.Companion.MULTIPLE_CHOICE_DIALOG_TYPE
+import mozilla.components.feature.session.prompt.ChoiceDialogFragment.Companion.MENU_CHOICE_DIALOG_TYPE
+import mozilla.components.feature.session.prompt.ChoiceDialogFragment.Companion.SINGLE_CHOICE_DIALOG_TYPE
+
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+internal const val FRAGMENT_TAG = "mozac_feature_prompt_dialog"
+
+/**
+ * Feature for displaying native dialogs for html elements like:
+ * input type date,file,time,color, option, menu, authentication, confirmation and other alerts.
+ *
+ * This feature will subscribe to the currently selected [Session] and display a suitable native dialog based on
+ * [Session.Observer.onPromptRequested] events. Once the dialog is closed or the user selects an item from the dialog
+ * the related [PromptFeature] will be consumed.
+ *
+ * @property sessionManager The [SessionManager] instance in order to subscribe to the selected [Session].
+ * @property fragmentManager The [FragmentManager] to be used when displaying a dialog (fragment).
+ */
+class PromptFeature(
```

Comment:
I wonder if we should move this into its own feature (like `feature-prompt`). Mostly because some of those prompt implementation that we are going to build may have more dependencies on other things and then we wouldn't want to add those to `browser-session`.

Reply:
Agree! Do you have an opinions @csadilek ?

Reply:
I moved it to its own feature, if you have any suggestions just let me know.